components/scream/src/share/io/scream_scorpio_interface.cpp

```diff
@@ -150,58 +176,39 @@ void grid_read_data_array(const std::string &filename, const std::string &varnam
 
 };
 /* ----------------------------------------------------------------- */
+// Handling the writing of output for packed arrays
+void grid_write_data_array(const std::string &filename, const std::string &varname, const std::vector<int>& dims, const Int& dim_length, const Int& pack_extra, const Real* hbuf)
+{
+  if (pack_extra == 0) // then no xtra data, is contiguous
+  {
+    grid_write_data_array_c2f_real_1d(filename.c_str(),varname.c_str(),dim_length,hbuf);
+  } else {
+    std::vector<Real> hbuf_new;
```

Comment:
Since this chunk of code is the same as in the `read` routine, I think it's best to factor it out into an independent function. Something like
```
void remove_padding (const Real* src, Real* dst, const std::vector<int>& phys_dims, const int padding);
```
and then call
```
int dim_length = std::accumulate(dims.begin(),dims.end(),1,std::multiplies<int>());
std::vector<Real> hbuf_new(dim_length);
remove_padding(hbuf, hbuf_new.data(),dims,pack_extra);
```